File: CT_BAEKJOON_8958.c
//https://www.acmicpc.net/problem/8958
//https://www.acmicpc.net/source/40298912

#include <stdio.h>
#pragma warning(disable:4996)
int main()
{
	int A, i, C, B, one, N, count,max ,min,ram;

	A = 0;
	B = 0;
	C = 0;
	N = 0;
	count = 0;
	max = 0;
	min = 0;
	one = 0;
	i = 0;
	ram = 0;



	scanf("%d", &N);
	
	for (int i = 0; i < N; i++) {
		char txt[80] = "";
		scanf("%s", txt);

		for (int i = 0; i < 80; i++) {
			
			if (txt[i] == 'O') {
				count = count + 1;
				//printf("%c is right ! \n" , txt[i]);
			}
			else {
				count = 0;
				//printf("wrong!\n");
			}
			ram = ram + count;
			//printf("ram is %d\n", ram);
		}
		printf("%d\n", ram);
		ram = 0;
		count = 0;
	}
}

File: CT_BAEKJOON_8958.py
#https://www.acmicpc.net/problem/8958
#https://www.acmicpc.net/source/40170943

"""
n = int(n)
a = []
total = 0
total = int(total)

a = list(map(int,input().split()))
"""

n = input()
n = int(n)
a = []
point = 1
point = int(point)
addpoint = 0
addpoint = int(addpoint)

for i in range(0 , n):
    A = input()
    for i in range(0 , len(A)):
        a.append(A[i:i+1])
    for i in range( 0 , len(A)):
        if a[i] == 'O':
            addpoint = point + addpoint
            point = point + 1

        else:
            point = 1
    print(addpoint)
    addpoint = 0
    point = 1
    a = []